File: src/main/java/com/heekng/celloct/dto/ManagerDto.java
package com.heekng.celloct.dto;

import com.heekng.celloct.entity.Manager;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

public class ManagerDto {

    @Getter
    public static class AddRequest {
        private Long shopId;
        private Long memberId;

        @Builder
        public AddRequest(Long shopId, Long memberId) {
            this.shopId = shopId;
            this.memberId = memberId;
        }
    }

    @Getter
    @Setter
    public static class ManagerResponse {
        private Long id;
        private String name;

        @Builder
        public ManagerResponse(Manager manager) {
            this.id = manager.getId();
            this.name = manager.getName();
        }
    }

    @Getter
    @Setter
    public static class WithMemberResponse {
        private Long managerId;
        private String managerName;
        private Long memberId;
        private String memberName;
        private String email;

        @Builder
        public WithMemberResponse(Manager manager) {
            this.managerId = manager.getId();
            this.managerName = manager.getName();
            this.memberId = manager.getMember().getId();
            this.memberName = manager.getMember().getName();
            this.email = manager.getMember().getEmail();
        }
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class UpdateRequest {
        private Long id;
        private String name;

        @Builder
        public UpdateRequest(Long id, String name) {
            this.id = id;
            this.name = name;
        }
    }

    @Getter
    @Setter
    public static class DeleteRequest {
        private Long managerId;
        private Long shopId;
        private Long memberId;

        @Builder
        public DeleteRequest(Long managerId, Long shopId, Long memberId) {
            this.managerId = managerId;
            this.shopId = shopId;
            this.memberId = memberId;
        }
    }

    @Getter
    @Setter
    public static class AddByStaffRequest {
        private Long shopId;
        private Long staffId;

        @Builder
        public AddByStaffRequest(Long shopId, Long staffId) {
            this.shopId = shopId;
            this.staffId = staffId;
        }
    }

    @Getter
    @Setter
    public static class AddByStaffResponse {
        private Long managerId;

        @Builder
        public AddByStaffResponse(Manager manager) {
            this.managerId = manager.getId();
        }
    }





}

File: src/main/kotlin/com/heekng/celloct/dto/ManagerDto.kt
package com.heekng.celloct.dto

import com.heekng.celloct.entity.Manager

class ManagerDto {
    data class AddRequest (
        val shopId: Long,
        val memberId: Long
    )

    data class ManagerResponse (val manager: Manager) {
        val id: Long?
        val name: String

        init {
            id = manager.id
            name = manager.name
        }
    }

    data class WithMemberResponse (val manager: Manager) {
        val managerId: Long?
        val managerName: String
        val memberId: Long?
        val memberName: String
        val email: String

        init {
            managerId = manager.id
            managerName = manager.name
            memberId = manager.member.id
            memberName = manager.member.name
            email = manager.member.email
        }
    }

    data class UpdateRequest (var id: Long?, val name: String)

    class DeleteRequest (
        val managerId: Long,
        val shopId: Long,
        val memberId: Long? = null
    )

    data class AddByStaffRequest (val shopId: Long, val staffId: Long)

    data class AddByStaffResponse (val manager: Manager) {
        val managerId: Long?

        init {
            managerId = manager.id
        }
    }
}